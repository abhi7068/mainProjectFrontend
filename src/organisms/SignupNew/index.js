import React, { useState, Fragment } from 'react';

import { Formik } from 'formik';

import * as Yup from 'yup';

import { connect } from 'react-redux';

import Grid from '@material-ui/core/Grid';

import { signUpApi } from '../../services/signUpServices';

import SignupTitle from '../../organisms/SignupTitle';
import FormikField from '../../molecules/FormikField';
import FormComponent from '../../atoms/FormComponent';
import ButtonOrange from '../../atoms/ButtonOrange';
import ButtonWhite from '../../atoms/ButtonWhite';

import PropTypes from 'prop-types';

import { Alert, AlertTitle } from '@material-ui/lab';
import { makeStyles } from '@material-ui/core/styles';

import { closeAuthUI } from '../../store/actions/authUIAction';

// styling organism, positioning atoms on Layout
const useStyles = makeStyles((theme) => ({
  positionImage: {
    position: 'absolute',
    bottom: '30px',
    left: '16px',
  },
  formContent: {
    padding: '16px 35px 16px',
    height: 'calc(100% - 26px - 16px)',
    position: 'relative',
  },
  link: {
    textDecoration: 'none',
  },
  gridTitle: {
    background: '#2874f0',
    padding: '40px 35px',
    color: 'white',
    borderTopLeftRadius: '4px',
    borderBottomLeftRadius: '4px',
  },
  root: {
    position: 'absolute',
    width: '58%',
    top: ' -12%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    zIndex: '10',
  },
}));

// Signup organism
const Signup = ({ openLogin, closeAuthUI }) => {
  const classes = useStyles();
  const [message, setMessage] = useState({ error: '', success: '' });

  // set validation rules for Signup Fields
  const [validationSchema] = useState(
    Yup.object().shape({
      name: Yup.string(
        'Enter full name as alphabets only, no numbers and special characters should be given'
      )
        .min(2, 'Name must contain at least 2 characters')
        .max(30, 'Name must contain no more then 20 characters')
        .required('Name is required'),
      email: Yup.string('Enter your email')
        .email('Enter a valid email')
        .required('Email is required')
        .max(35, 'Email must contain no more then 35 characters')
        .matches(
          /^[A-Za-z].*$/,
          'Email should not start with number orspecial character'
        ),
      confirmEmail: Yup.string('Enter your email')
        .required('Confirm your email')
        .oneOf([Yup.ref('email')], 'Email does not match')
        .max(35, 'Email must contain no more then 35 characters'),
      password: Yup.string('Enter a Password')
        .min(8, 'Password must contain at least 8 characters')
        .max(20, 'Password must contain no more then 20 characters')
        .required('Enter your password')
        .matches(
          /(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[\x21-\x2F\x3A-\x40\x5B-\x60\x7B-\x7E]).{8,20}/,
          'Password must contain at least one uppercase,number,special char'
        )
        .matches(/^[A-Z]/, 'Password must start with upper case'),
      confirmPassword: Yup.string('Enter a password')
        .required('Confirm your password')
        .oneOf([Yup.ref('password')], 'Enter correct password'),
    })
  );

  // set initial values for signup form
  const [initialValues] = useState({
    name: '',
    email: '',
    confirmEmail: '',
    password: '',
    confirmPassword: '',
  });

  // function which gets called when submit signup button is pressed
  const handleSignup = (values, resetForm) => {
    signUpApi(values).then((response) => {
      if (response && response.status === 200) {
        if (
          response.data[0] &&
          response.data[0].msg === 'E-mail already in use'
        ) {
          setMessage({ error: response.data[0].msg });
        } else {
          setMessage({ success: 'User signed up' });
          // resetForm({});
          // props.history.push('/login');
          setTimeout(() => openLogin(), 300);
        }
      } else {
        setMessage({ error: 'CANNOT SIGNUP' });
      }
    });
    // close popup
    // closeAuthUI()
  };
  return (
    <Fragment>
      <Grid item xs={5} className={classes.gridTitle}>
        <SignupTitle />
      </Grid>
      <Grid item xs={7}>
        <Fragment>
          {message && message.error && (
            <Alert severity="error" classes={{ root: classes.root }}>
              <AlertTitle>Error {message.error}</AlertTitle>
            </Alert>
          )}
          {message && message.success && (
            <Alert severity="success" classes={{ root: classes.root }}>
              <AlertTitle>Success {message.success}</AlertTitle>
            </Alert>
          )}
        </Fragment>

        <div className={classes.formContent}>
          <Formik
            initialValues={initialValues}
            onSubmit={(values, { resetForm }) => {
              handleSignup(values, resetForm);
            }}
            validationSchema={validationSchema}
          >
            {({ touched, errors }) => {
              return (
                <FormComponent noValidate>
                  <FormikField
                    name="name"
                    label="Name"
                    required
                    error={touched.name && Boolean(errors.name)}
                  />
                  <FormikField
                    name="email"
                    label="Email"
                    required
                    error={touched.email && Boolean(errors.email)}
                  />
                  <FormikField
                    name="confirmEmail"
                    label="Confirm email"
                    required
                    error={touched.confirmEmail && Boolean(errors.confirmEmail)}
                  />
                  <FormikField
                    name="password"
                    type="password"
                    label="Password"
                    error={touched.password && Boolean(errors.password)}
                    required
                  />
                  <FormikField
                    name="confirmPassword"
                    type="password"
                    label="Confirm Password"
                    error={
                      touched.confirmPassword && Boolean(errors.confirmPassword)
                    }
                    required
                  />

                  <ButtonOrange
                    variant="contained"
                    type="submit"
                    fullWidth
                    style={{ margin: '36px 0 16px 0' }}
                  >
                    Signup
                  </ButtonOrange>
                  <ButtonWhite
                    fullWidth
                    style={{ textTransform: 'none' }}
                    onClick={openLogin}
                  >
                    Existing User? Signin
                  </ButtonWhite>
                </FormComponent>
              );
            }}
          </Formik>
        </div>
      </Grid>
    </Fragment>
  );
};

Signup.propTypes = {
  closeAuthUI: PropTypes.func,
  openLogin: PropTypes.func,
};

export default connect(null, {
  closeAuthUI,
})(Signup);
