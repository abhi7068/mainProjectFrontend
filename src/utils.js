import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
