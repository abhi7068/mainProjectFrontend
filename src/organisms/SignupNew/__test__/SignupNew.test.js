import React from 'react';
import ReactDOM from 'react-dom';
import SignupNew from '../index';
import { render } from '../../../utils';
import { Provider } from 'react-redux';
import reducer from '../../../store/reducers/rootReducer';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
const store = createStore(reducer, applyMiddleware(thunk));
it('it renders', () => {
  render(<SignupNew />);
});

it('should take a snapshot', () => {
  const { asFragment } = render(<SignupNew />);
  expect(asFragment(render(<SignupNew />))).toMatchSnapshot();
});
it('renders with a expected length', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <SignupNew />
      </BrowserRouter>
    </Provider>,
    div
  );
  expect(div.querySelectorAll('input')).toHaveLength(5);
});
it('renders with a expected attribute', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <SignupNew />
      </BrowserRouter>
    </Provider>,
    div
  );
  expect(div.querySelector('input')).toHaveAttribute('type', 'text');
});
