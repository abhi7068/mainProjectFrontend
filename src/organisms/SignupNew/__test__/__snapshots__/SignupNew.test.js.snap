// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-gridTitle-256 MuiGrid-item MuiGrid-grid-xs-5"
  >
    <div
      class="makeStyles-leftTitle-361"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
        data-testid="Typography"
        style="font-size: 29px; font-weight: 500;"
      >
        Signup
      </h2>
      <p
        class="makeStyles-subtitle-393"
      >
        <span>
          We do not share your personal details with anyone.
        </span>
      </p>
      <div
        class="makeStyles-positionImage-362"
      >
        <img
          alt="login"
          src="yoyo-logo-mob.png"
          width="200"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
  >
    <div
      class="makeStyles-formContent-254"
    >
      <form
        action="#"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
          >
            Name
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input"
              name="name"
              required=""
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root Mui-required"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
          >
            Email
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input"
              name="email"
              required=""
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root Mui-required"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
          >
            Confirm email
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input"
              name="confirmEmail"
              required=""
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root Mui-required"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
          >
            Password
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input"
              name="password"
              required=""
              type="password"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root Mui-required"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
          >
            Confirm Password
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input"
              name="confirmPassword"
              required=""
              type="password"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root Mui-required"
          />
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained-460 makeStyles-buttonOrange-459 MuiButton-fullWidth"
          data-testid="ButtonOrange"
          style="margin: 36px 0px 16px 0px;"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Signup
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonWhite-493 MuiButton-fullWidth"
          style="text-transform: none;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Existing User? Signin
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
